### PR TITLE
Display error when processing a template fails.

### PR DESCRIPTION
**Pull request by csrwng**
Fixes message displayed when an error occurs and adds an alerts tag to the newfromtemplate page so it can actually display an error.

### CODE REVIEW

**Comment by csrwng** [comment]
@liggitt ptal when you get a chance

**Comment by smarterclayton** [comment]
or @spadgett

File: assets/app/scripts/controllers/newfromtemplate.js [hunk 1]
```diff
@@ -131,11 +131,15 @@ angular.module('openshiftConsole')
           Navigate.toProjectOverview($scope.projectName);
         },
         function(result) { // failure
+          var details;
+          if (result.data && result.data.message) {
```
Comment (liggitt) [inline assets/app/scripts/controllers/newfromtemplate.js]: do we always have `result.status`?

File: assets/app/scripts/controllers/newfromtemplate.js [hunk 2]
```diff
@@ -131,11 +131,15 @@ angular.module('openshiftConsole')
...
+            details = " " + result.data.message;
+          }
           $scope.alerts = [
             {
               type: "error",
               message: "An error occurred processing the template.",
-              details: "Status: " + result.status + ". " + result.data,
+              details: "Status: " + result.status + "." + details
```
Comment (liggitt) [inline assets/app/scripts/controllers/newfromtemplate.js]: what if `result.status` already ends in a `.`?

File: assets/app/scripts/controllers/newfromtemplate.js (hunk 1 above)
Comment (csrwng) [inline assets/app/scripts/controllers/newfromtemplate.js]: Status is the http status code, so I would say we do, but thinking about it, we probably don't need to display that here. If there are details, then we should show them and that's it.

File: assets/app/scripts/controllers/newfromtemplate.js (hunk 2 above)
Comment (csrwng) [inline assets/app/scripts/controllers/newfromtemplate.js]: It's only a numeric code, but as I said above, I think we can do without it.

**Comment by csrwng** [comment]
Updated details message for alert

**Comment by csrwng** [comment]
Fixes https://github.com/openshift/origin/issues/1775

File: assets/app/scripts/controllers/newfromtemplate.js [hunk 3]
```diff
@@ -131,11 +131,15 @@ angular.module('openshiftConsole')
           Navigate.toProjectOverview($scope.projectName);
         },
         function(result) { // failure
+          var details;
+          if (result.data && result.data.message) {
+            details = " " + result.data.message;
```
Comment (liggitt) [inline assets/app/scripts/controllers/newfromtemplate.js]: do we need the " " prefix now?

Comment (csrwng) [inline assets/app/scripts/controllers/newfromtemplate.js]: we don't. fixed.

**Comment by liggitt** [comment]
[merge]

**Comment by csrwng** [comment]
An error is happening on test-cmd.sh... rebasing

**Pull request merged**